test/test_nn.py

```diff
@@ -3728,6 +3728,13 @@ def forward(self, *args):
         module_name='PoissonNLLLoss',
         input_size=(2, 3, 4, 5),
         target_fn=lambda: torch.randn(2, 3, 4, 5).floor_().abs_(),
+        desc='non_full_loss',
+    ),
+    dict(
+        module_name='PoissonNLLLoss',
+        constructor_args=(True, False, True, 1e-8, False),
+        input_size=(2, 3, 4, 5),
+        target_fn=lambda: torch.randn(2, 3, 4, 5).floor_().abs_(),
```

Comment:
Could you add a `reference_fn` for this and the `no_reduce` (`non_full_loss`) case? See `kldivloss_no_reduce_test` [here](https://github.com/pytorch/pytorch/pull/3330/files#diff-d89baec73022f5f511c5beb5ce6498dfR3677) for an example